products/views.py

```diff
@@ -29,3 +36,111 @@ def home_page(request):
     }
 
     return render(request, "product/dashboard.html", context)
+
+
+@login_required
+def add_brand(request):
+    """ To add a new brand in brand model """
+
+    if not request.user.is_superuser:
+        return Http404
+
+    context = {}
+    if request.method == "POST":
+        brand = AddBrandForm(request.POST, request.FILES)
+        if brand.is_valid():
+            brand.save()
+            messages.success(request, BrandFormSuccessMessages.NEW_BRAND_ADDED)
+        return redirect('view-brand')
+
+    brand_form = AddBrandForm()
+
+    context = {
+        "form": brand_form,
+        "heading": AdminPortalHeadings.ADD_BRAND
+    }
+    return render(request, "product/brand/add_brand.html", context)
+
+
+@login_required
+def update_brands(request, pk):
+    """ To update brand details """
+
+    if not request.user.is_superuser:
+        return Http404
+
+    context = {}
+
+    if request.method == "POST":
+        brand_form = UpdateBrandForm(
+            request.POST, request.FILES, instance=get_object_or_404(Brand, id=pk)
+        )
+        if brand_form.is_valid():
+            brand_form.save()
+            messages.success(request, BrandFormSuccessMessages.BRAND_UPDATED)
+        return redirect('view-brand')
+
+    brand_form = UpdateBrandForm(instance=Brand.objects.get(id=pk))
```

Comment:
@rinkishivani there is no need to do multiple query for Brand instance. you can query one time and assign it to a variable and then use that variable in `UpdateBrandForm(instance=Brand.objects.get(id=pk))`

cc: @lalitvasoya

Reply:
Done.